src/v/cluster/service.cc

```diff
@@ -679,6 +679,27 @@ ss::future<transfer_leadership_reply> service::transfer_leadership(
     }
 }
 
+ss::future<producer_id_lookup_reply> service::highest_producer_id(
+  producer_id_lookup_request&&, rpc::streaming_context&) {
+    producer_id_lookup_reply reply;
+    auto highest_pid = co_await _partition_manager.map_reduce0(
+      [](const partition_manager& pm) {
+          model::producer_id pid{};
+          for (const auto& [_, p] : pm.partitions()) {
+              pid = std::max(pid, p->highest_producer_id());
+          }
```

Comment:
It's a good question, and the answer is pretty much the same as the question I posed in the cover letter:

>Why can't we just upload a single producer ID from the id_allocator when uploading metadata? Segment uploads are not coordinated by any central entity. This means that if there were a central uploader that queried the highest producer ID from the id_allocator, there would be a strong chance that new transactional segments get initialized and uploaded before the next ID is included in uploaded metadata.

Reply:
if i understand that correctly nothing will prevent a new request  allocating new producer id to be processed by any of the partition after this full scan. Would that be the same as in the case we check only the `id_allocator` for the highest producer_id ?

Reply:
>Would that be the same as in the case we check only the id_allocator for the highest producer_id ?

Right, exactly. Because all uploads are disjoint with no stop-the-world kind of sync point, there's always the chance for a window where more producer_ids may be used before the next id_allocator upload (whether that's uploading/scanning the entire partition, or just snapshotting the lastest id state of `id_allocator_stm`).

Reply:
>pretty much the same as the question I posed in the cover letter:

ahh missed th